nixos/modules/services/misc/matrix-conduit.nix

```diff
@@ -0,0 +1,233 @@
+          When this is enabled, ACME will be used to retrieve a TLS certificate by default. To disable
+          this, set the <option>services.nginx.virtualHosts.&lt;settings.global.server_name&gt;.enableACME</option> to
+          <literal>false</literal> and if appropriate do the same for
+          <option>services.nginx.virtualHosts.&lt;settings.global.server_name&gt;.forceSSL</option>.
+        '';
+      };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.matrix-conduit;
+        defaultText = "pkgs.matrix-conduit";
+        example = "pkgs.matrix-conduit";
+        description = ''
+          Package of the conduit matrix server to use.
+        '';
+      };
+
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = format.type;
+          options = {
+            global.server_name = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "example.com";
+              description = "The server_name is the name of this server. It is used as a suffix for user # and room ids.";
+            };
+            global.port = mkOption {
+              type = types.port;
+              default = 6167;
+              description = "The port Conduit will be running on. You need to set up a reverse proxy in your web server (e.g. apache or nginx), so all requests to /_matrix on port 443 and 8448 will be forwarded to the Conduit instance running on this port";
+            };
+            global.max_request_size = mkOption {
+              type = types.ints.positive;
+              default = 20000000;
+              description = "Max request size in bytes. Don't forget to also change it in the proxy.";
+            };
+            global.allow_registration = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new users can register on this server.";
+            };
+            global.allow_encryption = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new encrypted rooms can be created. Note: existing rooms will continue to work.";
+            };
+            global.allow_federation = mkOption {
+              type = types.bool;
+              default = false;
+              description = ''
+                Whether this server federates with other servers.
+                Upstream documentation contains the following warning:
+                You should not join Matrix rooms without asking the admins first.
+                We do not know whether Conduit is safe for general use yet, so you
+                should assume there is some chance that it breaks rooms permanently
+                for all participating users. We are not aware of such a bug today,
+                but we would like to do more testing.
+              '';
+            };
+            global.allow_jaeger = mkOption {
```

Comment:
I think RFC 42 states that this pattern should primarily be used for configuration options that are very commonly set. Of course the line between important and less important options is not clear cut, but I'd argue that you could remove at least `allow_jaeger`, `max_request_size` and maybe also `trusted_servers` and `address`

Reply:
`max_request_size` will be set sooner or later as soon as somebody tries to upload a video :joy: . `address` will be touched as soon as somebody tries to use the nixos container system. I like explicitly exposing `trusted_servers` because it's a security critical variable and it should at least be explicitly set in the toml for the sake of transparency.
I have no feelings about dropping or keeping `allow_jaeger`.